### PR TITLE
Add `/api/v1/configuration` endpoint to get available locales

### PR DESCRIPTION
**Pull request by rlustin**
Availables locales are stored in an environment variable.

```
curl -v -H "Authorization: Bearer goodtokenfortests" -H "Content-Type: application/json"  http://127.0.0.1:3100/api/v1/configuration
> GET /api/v1/configuration HTTP/1.1
> Host: 127.0.0.1:3100
> User-Agent: curl/7.51.0
> Accept: */*
> Authorization: Bearer goodtokenfortests
> Content-Type: application/json
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Content-Length: 38
< Date: Sat, 14 Jan 2017 14:48:49 GMT
{"locales":["de","en","es","fr","it"]}% 
```

Also move API routing in a dedicated file.

### CODE REVIEW

File: src/api/v1/configuration.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#[derive(RustcEncodable)]
+struct Configuration {
+    locales: Vec<String>,
+}
+
+pub fn index(_: &mut Request) -> IronResult<Response> {
+    dotenv().ok();
+
+    let locales: Vec<String> = env::var("LUGH_AVAILABLE_LOCALES")
+        .expect("LUGH_AVAILABLE_LOCALES must be set")
```
Comment (Signez) [inline src/api/v1/configuration.rs]: Hm. Shouldn't this kind of tests be done on application startup?

**Comment by rlustin** [comment]
Thanks for the review!

You are definitely right, this is a temporary hack to quickly expose you this endpoint to implement the add of new translations on the frontend ;) If you need it right away, you can merge it as is. If you’re not in a hurry, I’ll work on this in the following days.

Let’s discuss how to store these configuration settings in the database by chat anyway!

File: src/api/v1/configuration.rs (hunk 1 above)
Comment (rlustin) [inline src/api/v1/configuration.rs]: Yes, but as it’s a temporary hack  ¯\_(ツ)_/¯

**Comment by Signez** [comment]
> You are definitely right, this is a temporary hack to quickly expose you this endpoint to implement the add of new translations on the frontend ;) If you need it right away, you can merge it as is. If you’re not in a hurry, I’ll work on this in the following days.

Cool! I'm okay with merging this thing for the moment as soon as my comment is resolved (about checking this ENV variable on application startup).

File: src/api/v1/configuration.rs (hunk 1 above)
Comment (rlustin) [inline src/api/v1/configuration.rs]: Fixed.

**Pull request merged**